app/design/frontend/Magento/storefront/Magento_Cms/layout/cms_index_index.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+    <body>
+        <referenceContainer name="content">
+            <block class="Magento\Framework\View\Element\Template" name="main-baner"/>
+        </referenceContainer>
```

Comment:
Everything is correct there, but you need to create your banner in admin as clients might want to edit the Homepage:
1. Move your banner html from app/design/frontend/Magento/storefront/Magento_Cms/templates/html/banner.phtml  to Content -> CMS Blocks
2. Then include your CMS block in the CMS page homepage
3. Delete your template from the code and also delete layout instructions